astropy/io/fits/scripts/fitsheader.py

```diff
@@ -275,9 +292,76 @@ def print_headers_as_table(args):
     resulting_table.write(sys.stdout, format=args.table)
 
 
+def print_headers_as_comparison(args):
+    """Prints FITS header(s) with keywords as columns.
+    
+    This follows the dfits+fitsort format.
+
+    Parameters
+    ----------
+    args : argparse.Namespace
+        Arguments passed from the command-line as defined below.
+    """
+    tables = []
+    # Create a Table object for each file
+    for filename in args.filename:  # Support wildcards
+        try:
+            formatter = TableHeaderFormatter(filename)
+            tbl = formatter.parse(args.extensions,
+                                  args.keywords,
+                                  args.compressed)
+            if tbl:
+                tables.append(tbl)
+        except OSError as e:
+            log.error(str(e))  # file not found or unreadable
+    # Concatenate the tables
+    if len(tables) == 0:
+        return False
+    elif len(tables) == 1:
+        resulting_table = tables[0]
+    else:
+        from .... import table
+        resulting_table = table.vstack(tables)
+    # If we obtained more than one hdu, merge hdu and keywords columns
+    if len(np.unique(resulting_table['hdu']))>1:
+        for tab in tables:
+            new_column = table.Column(['{}:{}'.format(row['hdu'],row['keyword']) for row in tab])
```

Comment:
`table` is not defined if printing keywords for multiple extensions with only one file (`fitsheader --fitsort -k EXPSTART -k EXPTIME -k ESO.DET.ID astropy/io/fits/tests/data/test1.fits`).